android/lib/src/main/java/com/swmansion/gesturehandler/GestureHandlerOrchestrator.kt

```diff
@@ -329,19 +320,71 @@ class GestureHandlerOrchestrator(
     return parent === wrapperView
   }
 
-  private fun extractCoordsForView(view: View?, event: MotionEvent, outputCoords: FloatArray) {
-    if (view === wrapperView) {
-      outputCoords[0] = event.x
-      outputCoords[1] = event.y
-      return
+  /**
+   * Transforms an event in the coordinates of wrapperView into the coordinate space of the received view.
+   *
+   * This modifies and returns the same event as it receives
+   *
+   * @param view - view to which coordinate space the event should be transformed
+   * @param event - event to transform
+   */
+  fun transformEventToViewCoords(view: View?, event: MotionEvent): MotionEvent {
```

Comment:
not sure about the naming of this method. It suggests that it mutates the provided object while sometimes it doesn't. Also, it looks like it mutates the event in a very few scenarios. Unless I don't understand this well it appears like it only mutates if the view is the wrapperView (which happens only at the very top of the view hierarchy) and when view is `null` which I don't think should ever happen?

Reply:
It traverses the view hierarchy upwards until it finds `wrapperView` (which is an instance of `GestureHandlerRootView`) or `null` (in case there is no root view we don't want to crash the app) and then comes back recursively transforming the event using the same matrices that are used to transform views.

It will not mutate the provided event in the following cases:
- `null` is passed as an argument for `view`
- all the views on the path from the `wrapperView` to the provided view aren't transformed (their matrix is identity) and are positioned at the top-left corner of their parents

As for the name, I wanted to make It obvious that it doesn't return a new object but may transform the one provided as an argument so it's not safe to use if the untransformed event is also needed. Semantically I believe it's fine since the only case where the event is not transformed is when the coordinate space of the provided view is the same as the wrapper view.